src/rules/noUnusedVariableRule.ts

```diff
@@ -351,37 +355,71 @@ function findImport(pos: number, sourceFile: ts.SourceFile): ts.Identifier | und
 
             const { name: defaultName, namedBindings } = i.importClause;
             if (namedBindings !== undefined && namedBindings.kind === ts.SyntaxKind.NamespaceImport) {
-                const { name } = namedBindings;
-                if (name.getStart() === pos) {
-                    return name;
-                }
-                return undefined;
+                return [namedBindings.name];
             }
 
-            if (defaultName !== undefined && defaultName.getStart() === pos) {
-                return defaultName;
+            // Starting from TS2.8, when all imports in an import node are not used,
+            // TS emits only 1 diagnostic object for the whole line as opposed
+            // to the previous behavior of outputting a diagnostic with kind == 6192
+            // (UnusedKind.VARIABLE_OR_PARAMETER) for every unused import.
+            // From TS2.8, in the case of none of the imports in a line being used,
+            // the single diagnostic TS outputs are different between the 1 import
+            // and 2+ imports cases:
+            // - 1 import in node:
+            //   - diagnostic has kind == 6133 (UnusedKind.VARIABLE_OR_PARAMETER)
+            //   - the text range is the whole node (`import { ... } from "..."`)
+            //     whereas pre-TS2.8, the text range was for the import node. so
+            //     `name.getStart()` won't equal `pos` like in pre-TS2.8
+            // - 2+ imports in node:
+            //   - diagnostic has kind == 6192 (UnusedKind.DECLARATION)
+            //   - we know that all of these are unused
+            if (kind === UnusedKind.DECLARATION) {
+                const imp: ts.Identifier[] = [];
+                if (defaultName !== undefined) {
+                    imp.push(defaultName);
+                }
+                if (namedBindings !== undefined) {
+                    imp.push(...namedBindings.elements.map((el) => el.name));
+                }
+                return imp.length > 0 ? imp : undefined;
+            } else if (defaultName !== undefined && (
+                isInRange(defaultName, pos) || namedBindings === undefined // defaultName is the only option
+            )) {
+                return [defaultName];
             } else if (namedBindings !== undefined) {
-                for (const { name } of namedBindings.elements) {
-                    if (name.getStart() === pos) {
-                        return name;
+                if (namedBindings.elements.length === 1) {
+                    return [namedBindings.elements[0].name];
+                }
+
+                for (const element of namedBindings.elements) {
+                    if (isInRange(element, pos)) {
+                        return [element.name];
                     }
                 }
             }
         }
         return undefined;
     });
+    return imports !== undefined ? imports : [];
+}
+
+function isInRange(range: ts.TextRange, pos: number): boolean {
```

Comment:
Nit: might belong in some utils file somewhere? `isInTextRange`?

Reply:
hmm not sure where it would belong though